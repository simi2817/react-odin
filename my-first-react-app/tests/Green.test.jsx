import { describe, it, expect } from "vitest";
import { render, screen } from '@testing-library/react';
import Green from '../src/components/test/Green';
import userEvent from "@testing-library/user-event";

describe('Green', () => {
    it('renders correct heading', () => {
        render(<Green />);
        expect(screen.getByRole("heading").textContent).toMatch(/magnificent monkeys/i);
    });
});

describe('Green Component', () => {
    it('renders magnificent monkeys', () => {
        //// since screen does not have the container property, we'll destructure render to obtain a container for this test
        const { container } = render(<Green />);
        expect(container).toMatchSnapshot();
    });

    it('renders radical rhinos after button click', async() => {
        const user = userEvent.setup();
        render(<Green />);
        const button = screen.getByRole("button", { name: "Click Me" });
        await user.click(button);
        expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
        screen.debug();
    });
});