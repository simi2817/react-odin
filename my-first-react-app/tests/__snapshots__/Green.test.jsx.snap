// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Green Component > renders magnificent monkeys 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;
